### ISSUE TITLE
[bug] Convert into transfer: Erroneous confirmation dialog in case an income is converted

### ISSUE THREAD

**Issue by xeddicuz** [mtotschnig/MyExpenses#1514]
Specs:

    Samsung A14
    App Version: 3.8.5 (Revision 739) 20240531-0532 GITHUB org.fdroid.basic

When using _convert into transfer_ the origin/destination accounts are swapped.

Steps to reproduce.
1. Select a transfer in account **A**.
1a. Convert into transfer.
2. A dialog "Convert into transfer" appears.
2a. Select account **B**.
2b. Click button: "To **B**".
3. A dialog appears: "The selected transfer will be replaced by a transfer from **A** to **B**".
3a. Press OK.

Result(bug): a dialog appears: "Transfer: From B To A". 

_Now you have to press Edit and swap the accounts to achieve the original intent._

Kind regards
David.

**Comment by mtotschnig** [comment]
@xeddicuz Thank you for the report. The "Convert into transfer" feature is supposed to maintain the sign of the transaction that is converted.
So I think the bug is not that the accounts are swapped in the result, but that the confirmation dialog shows wrong information, but only if the transaction is an income:
If you convert an expense in account A, then select account B, the dialog says "The selected transaction will be replaced by a transfer from A to B", which is correct because the money will go from A to B.
If you convert an income in account A, then select account B, the dialog currently also says "The selected transaction will be replaced by a transfer from A to B". The correct message would be "The selected transaction will be replaced by a transfer from B to A".

**Comment by xeddicuz** [comment]
@mtotschnig Thanks for figuring it out 👍

**Comment by mtotschnig** [comment]
@xeddicuz My Expenses is not a double entry book-keeping system. In My Expenses a transaction is linked to one account, and either has a positive or negative amount. A transfer is actually a pair of two linked transactions recorded for two different accounts, where the amount is positive for one and negative for the other. If it is not a foreign currency transfer, normally the amount on one side would be exactly the negative of the amount on the other side.
What the "convert to transfer" command does, is to create a counterpart for a transaction in a second account, and link the two as transfer. 
One use case for this is the following: You import transactions from your bank, and one transaction is an ATM withdrawal, so the money actually went to your wallet, and you also track this wallet with My Expenses. By converting the transaction to a transfer, you will get the corresponding entry for your wallet, that represents the increase of your balance by the amount you withdraw.

**Comment by xeddicuz** [comment]
Exactly, so when you convert a transaction in (a) Rekening "(b)To wallet". It shouldnt be from (b)wallet to (a)rekening.
Which would mean the dialog is swapped like I stated in the first message.

let's assume I imported the transaction (-455.44) from Rekening and I manually deposited (+455.44) it to the Wallet account(let's say Wallet is an actual bank account, which is the case for me, it's my weekly budget account with a card. (I use a Cash account for my cash, i didnt name Wallet properly.)) from my bank.

I would assume the transaction could be converted to a transfer which would decrease the balance from Rekening(-455.44), like a transaction had, and increase the balance on Wallet(+455.44) to signify what happened, in My Expenses. Else why convert a transaction to transfer in the first place?

If this is not the case I need to learn more about financing..
